http/handler.go

```diff
@@ -58,7 +58,7 @@ func Handler(core *vault.Core) http.Handler {
 	mux.Handle("/v1/sys/wrapping/unwrap", handleRequestForwarding(core, handleLogical(core, false, wrappingVerificationFunc)))
 	mux.Handle("/v1/sys/capabilities-self", handleRequestForwarding(core, handleLogical(core, true, nil)))
 	mux.Handle("/v1/sys/", handleRequestForwarding(core, handleLogical(core, true, nil)))
-	mux.Handle("/v1/", handleRequestForwarding(core, handleLogical(core, false, nil)))
+	mux.Handle("/v1/", handleCORS(core, handleRequestForwarding(core, handleLogical(core, false, nil))))
```

Comment:
That was my original thought, but as I started to consider things I felt that the sys path should remain inaccessible by cross-origin requests. As always I'm happy to refactor if you all feel differently. 

I certainly can imagine the desire to have /sys serving CORS headers if someone wanted to develop a web UI for Vault, but to me adding a new handler for a separate path that exposes only /sys functions that make sense is a more secure way of doing things.

Although CORS headers do not change anything with Vault, just allow browsers to talk to it directly from a different origin.